### PR TITLE
Opal 146 sharing options can be specified in android client

### PR DESCRIPTION
**Pull request by sznemeth3**

### CODE REVIEW

File: app/src/main/java/com/greenfox/kalendaryo/adapter/SharingOptionsAdapter.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    public class ViewHolder extends RecyclerView.ViewHolder {
+        private TextView calendarName;
+        private Spinner dropdown;
+
+
+        public ViewHolder(View itemView) {
+            super(itemView);
+            calendarName = itemView.findViewById(R.id.calendarname);
+            dropdown = (Spinner)itemView.findViewById(R.id.set_visibility);
+            String[] items = new String[]{"Default visibility", "Public", "Private"};
```
Comment (arnikaeva) [inline app/src/main/java/com/greenfox/kalendaryo/adapter/SharingOptionsAdapter.java]: use enum

File: app/src/main/java/com/greenfox/kalendaryo/adapter/SharingOptionsAdapter.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+import com.greenfox.kalendaryo.models.Kalendar;
+
+import java.util.List;
+
+
+public class SharingOptionsAdapter extends RecyclerView.Adapter<SharingOptionsAdapter.ViewHolder> {
+
+    private List<GoogleCalendar> inputCalendarIds;
+    private Context context;
+    Kalendar kalMerged;
```
Comment (arnikaeva) [inline app/src/main/java/com/greenfox/kalendaryo/adapter/SharingOptionsAdapter.java]: naming

File: app/src/main/java/com/greenfox/kalendaryo/SharingOptionsActivity.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+                        recyclerLayoutManager.getOrientation());
+        recKal.addItemDecoration(dividerItemDecoration);
+        goToChooseAccount = findViewById(R.id.gotochooseaccount);
+        goToChooseAccount.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Intent i = new Intent(SharingOptionsActivity.this, ChooseAccountActivity.class);
+                i.putExtra("list", kalMerged);
+                startActivity(i);
+                adapter.removeInputCalendarIds();
```
Comment (arnikaeva) [inline app/src/main/java/com/greenfox/kalendaryo/SharingOptionsActivity.java]: why are we removing this?

File: app/src/main/java/com/greenfox/kalendaryo/SharingOptionsActivity.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        Kalendar kalMerged = (Kalendar) getIntent().getSerializableExtra("list");
+        setContentView(R.layout.activity_sharing_options);
+
+        adapter = new SharingOptionsAdapter(this, kalMerged);
+
+        RecyclerView recKal = findViewById(R.id.sharing_options_list);
```
Comment (arnikaeva) [inline app/src/main/java/com/greenfox/kalendaryo/SharingOptionsActivity.java]: naming

File: app/src/main/java/com/greenfox/kalendaryo/SharingOptionsActivity.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+public class SharingOptionsActivity extends AppCompatActivity {
+
+    private SharingOptionsAdapter adapter;
+    Button goToChooseAccount;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        Kalendar kalMerged = (Kalendar) getIntent().getSerializableExtra("list");
```
Comment (arnikaeva) [inline app/src/main/java/com/greenfox/kalendaryo/SharingOptionsActivity.java]: naming for `kalMerged` and for `list`

File: app/src/main/res/layout/activity_select_calendar.xml [hunk 6]
```diff
@@ -16,7 +16,7 @@
         app:layout_constraintStart_toStartOf="parent"/>
 
     <Button
-        android:id="@+id/gotochooseaccount"
+        android:id="@+id/gotosharingoptions"
```
Comment (arnikaeva) [inline app/src/main/res/layout/activity_select_calendar.xml]: for resource naming, check:
https://github.com/umesh0492/android-guidelines/blob/master/project_and_code_guidelines.md
**2.3.2 Resources naming**

**Review by arnikaeva — CHANGES_REQUESTED** [review]
Styleguide related fixes :)

File: app/src/main/java/com/greenfox/kalendaryo/SelectCalendarActivity.java [hunk 7]
```diff
@@ -56,27 +57,24 @@ protected void onCreate(Bundle savedInstanceState) {
         kalendar = new Kalendar();
         getCalendarList();
         adapter.setListChange(kalendar);
-        recKal = findViewById(R.id.listView);
-        recKal.setAdapter(adapter);
+        recyclerView = findViewById(R.id.listView);
```
Comment (bekobarna) [inline app/src/main/java/com/greenfox/kalendaryo/SelectCalendarActivity.java]: why are we calling it a recyclerview and than call it a listview in the xml?

File: app/src/main/java/com/greenfox/kalendaryo/SharingOptionsActivity.java [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+        setContentView(R.layout.activity_sharing_options);
+
+        adapter = new SharingOptionsAdapter(this, kalendar);
+
+        RecyclerView recyclerView = findViewById(R.id.sharing_options_list);
+        recyclerView.setAdapter(adapter);
+        LinearLayoutManager recyclerLayoutManager = new LinearLayoutManager(this);
+        recyclerView.setLayoutManager(recyclerLayoutManager);
+        DividerItemDecoration dividerItemDecoration =
+                new DividerItemDecoration(recyclerView.getContext(),
```
Comment (bekobarna) [inline app/src/main/java/com/greenfox/kalendaryo/SharingOptionsActivity.java:33]: indentation

File: app/src/main/java/com/greenfox/kalendaryo/adapter/GoogleCalendarAdapter.java [hunk 9]
```diff
@@ -95,7 +95,7 @@ public ViewHolder(View itemView) {
 
     }
     public interface ListChange {
-        void saveCalendar(String calendarTitle);
-        void removeCalendar(String calderTitle);
+        void saveCalendar(GoogleCalendar calendarTitle);
+        void removeCalendar(GoogleCalendar calderTitle);
```
Comment (bekobarna) [inline app/src/main/java/com/greenfox/kalendaryo/adapter/GoogleCalendarAdapter.java]: caldertitle?

File: app/src/main/java/com/greenfox/kalendaryo/adapter/SharingOptionsAdapter.java [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+        public ViewHolder(View itemView) {
+            super(itemView);
+            calendarName = itemView.findViewById(R.id.textView_calendarname);
+            dropdown = (Spinner)itemView.findViewById(R.id.spinner_setVisibility);
+            String[] items = new String[]{"Default visibility", "Public", "Private"};
+            ArrayAdapter<VisibilityOptions> adapter = new ArrayAdapter<VisibilityOptions>(context, android.R.layout.simple_spinner_dropdown_item, VisibilityOptions.values());
+            dropdown.setAdapter(adapter);
+        }
+    }
+}
```
Comment (bekobarna) [inline app/src/main/java/com/greenfox/kalendaryo/adapter/SharingOptionsAdapter.java]: put an empty line at the end

File: app/src/main/java/com/greenfox/kalendaryo/adapter/SharingOptionsAdapter.java [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+
+import java.util.List;
+
+
+public class SharingOptionsAdapter extends RecyclerView.Adapter<SharingOptionsAdapter.ViewHolder> {
+
+    private List<GoogleCalendar> googleCalendars;
+    private Context context;
+    Kalendar kalendar;
+    public enum VisibilityOptions{
```
Comment (arnikaeva) [inline app/src/main/java/com/greenfox/kalendaryo/adapter/SharingOptionsAdapter.java]: do not use plural for enum names, `VisibilityOption` is fine

File: app/src/main/java/com/greenfox/kalendaryo/models/GoogleCalendar.java [hunk 12]
```diff
@@ -3,14 +3,13 @@
...
- * Created by barba on 04/01/2018.
- */
+import java.io.Serializable;
 
-public class GoogleCalendar implements Parcelable {
+public class GoogleCalendar implements Parcelable, Serializable {
 
     String id;
     String summary;
+    String sharingOptions;
```
Comment (arnikaeva) [inline app/src/main/java/com/greenfox/kalendaryo/models/GoogleCalendar.java]: type for sharingOption should be VisibilityOption enum, so it cannot take other value

**Review by arnikaeva — CHANGES_REQUESTED** [review]
Minor comments

**Review by bekobarna — DISMISSED** [review]

File: app/src/main/java/com/greenfox/kalendaryo/adapter/SharingOptionsAdapter.java [hunk 13]
```diff
@@ -0,0 +1,90 @@
...
+        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.sharing_options, parent, false);
+        SharingOptionsAdapter.ViewHolder viewHolder = new SharingOptionsAdapter.ViewHolder(view);
+        return viewHolder;
+    }
+
+    @Override
+    public void onBindViewHolder(SharingOptionsAdapter.ViewHolder holder, int position) {
+        GoogleCalendar calendar = googleCalendars.get(position);
+        holder.calendarName.setText(calendar.getSummary());
+        calendar.setSharingOption(VisibilityOption.valueOf("DEFAULT"));
```
Comment (arnikaeva) [inline app/src/main/java/com/greenfox/kalendaryo/adapter/SharingOptionsAdapter.java]: I think you can access it like VisibilityOption.DEFAULT

File: app/src/main/java/com/greenfox/kalendaryo/adapter/SharingOptionsAdapter.java [hunk 14]
```diff
@@ -0,0 +1,90 @@
...
+
+import java.util.List;
+
+
+public class SharingOptionsAdapter extends RecyclerView.Adapter<SharingOptionsAdapter.ViewHolder> {
+
+    private List<GoogleCalendar> googleCalendars;
+    private Context context;
+    Kalendar kalendar;
+    public enum VisibilityOption{
```
Comment (arnikaeva) [inline app/src/main/java/com/greenfox/kalendaryo/adapter/SharingOptionsAdapter.java]: move it to its own file (VisibilityOption.java)

**Review by arnikaeva — CHANGES_REQUESTED** [review]

**Review by bekobarna — APPROVED** [review]

**Review by arnikaeva — APPROVED** [review]

**Pull request merged**